dsl/ChalkODMProp.h

```diff
@@ -7,38 +7,39 @@ namespace sorbet::dsl {
 /**
  * This class desugars things of the form
  *
- *   prop :foo, String
+ *   prop :foo, Type
  *
  * into
  *
- *   sig {returns(T.nilable(String))}
- *   def foo; T.cast(nil, T.nilable(String)); end
- *   sig {params(arg0: String).returns(NilClass)}
- *   def foo=(arg0); end
+ *   sig {returns(Type)}
+ *   def foo; ...; end
+ *   sig {params(x: Type).returns(Type)}
+ *   def foo=(x); ...; end
  *   class Mutator < Chalk::ODM::Mutator
- *     sig {params(arg0: String).returns(NilClass)}
- *     def foo=(arg0); end
+ *     sig {returns(Type)}
+ *     def foo(x); end
```

Comment:
Do we synthesize a getter in the Mutator? And if we do, do we synthesize it with an arg `x`?

Reply:
There's conditional logic to sometimes synthesize a mutator, and if the mutator is synthesized to sometimes have it have a getter. It looks like the mutator doesn't take any arguments, though, that's my mistake.